diff --git a/pathauto.module b/pathauto.module
index b4bdb32..f830c2e 100644
--- a/pathauto.module
+++ b/pathauto.module
@@ -757,3 +757,47 @@ function pathauto_blog_update_alias(stdClass $account, $op, array $options = arr
 /**
  * @} End of "name pathauto_user".
  */
+
+/**
+ * Implements hook_action_info().
+ */
+function pathauto_action_info() {
+  return array(
+    'pathauto_node_update_action' => array(
+      'type' => 'node',
+      'label' => t('Update node alias'),
+      'configurable' => FALSE,
+    ),
+    'pathauto_user_update_action' => array(
+      'type' => 'user',
+      'label' => t('Update user alias'),
+      'configurable' => FALSE,
+    ),
+    'pathauto_taxonomy_term_update_action' => array(
+      'type' => 'taxonomy_term',
+      'label' => t('Update taxonomy term alias'),
+      'configurable' => FALSE,
+    ),
+  );
+}
+
+/**
+ * Update action wrapper for pathauto_node_update_alias().
+ */
+function pathauto_node_update_action($node, $context = array()) {
+  pathauto_node_update_alias($node, 'bulkupdate', array('message' => TRUE));
+}
+
+/**
+ * Update action wrapper for pathauto_user_update_alias().
+ */
+function pathauto_user_update_action($user, $context = array()) {
+  pathauto_user_update_alias($user, 'bulkupdate', array('message' => TRUE));
+}
+
+/**
+ * Update action wrapper for pathauto_taxonomy_term_update_alias().
+ */
+function pathauto_taxonomy_term_update_action($term, $context = array()) {
+  pathauto_taxonomy_term_update_alias($term, 'bulkupdate', array('message' => TRUE));
+}
\ No newline at end of file
